### ISSUE TITLE
Feature Request: Add setting to automatically reorder checkboxes.

### ISSUE THREAD

**Issue by TakeADev** [Erl-koenig/obsidian-checkboxReorder#6]
This feature will create a settings tab where users can toggle the ability to automatically reorder checkboxes when a file is modified. 

Expected behavior:  When user "checks" a checkbox, the checklist item will be automatically placed at the bottom of the list.

**Comment by TakeADev** [comment]
I've already created and implemented the feature through my own fork. Let me know if you would like me to create a pr for this!

**Comment by Erl-koenig** [comment]
Hi Chance, great idea! You can create a PR for this, I'll have a look :)

**Comment by TakeADev** [comment]
> Hi Chance, great idea! You can create a PR for this, I'll have a look :)

Awesome! While writing the PR, I noticed some bugs with the implementation...Let me see if I can iron those out before submitting lol.

**Comment by lukemt** [comment]
instead of doing this on every file change, it might be a good idea to only run this when a check-box was toggled. not sure if there is any advanced event system in obsidian, but i hope one can listen for commands, this would be necessary. additionally one could listen for click events and check if the `currentTarget` is a checkbox. I could help with the second one but the first...?

**Comment by TakeADev** [comment]
Sorry, I created my PR right as you commented.

I tried looking for some sort of better event handler but wasn't able to find anything myself. That would be great if you could help find a better solution. I'm also not 100% satisfied with the solution I found.

**Comment by lukemt** [comment]
Yeah I don't know if one can listen to commands...

**Comment by lukemt** [comment]
otherwise the if statement may be wise

**Comment by lukemt** [comment]
i mean if it is behind a setting.. and it's not really doing much, it may be fine

**Comment by lukemt** [comment]
do you toggle checkboxes with the mouse or via commands?

I was thinking one could add a command for toggeling a checkbox AND reordering it.

Just an idea

**Comment by TakeADev** [comment]
> i mean if it is behind a setting.. and it's not really doing much, it may be fine

This was my thinking as well. Is it ideal? Not really...If a better solution can be found, that'd be great, but I can't imagine it making that much of a performance difference. 

> do you toggle checkboxes with the mouse or via commands?

I personally just use the mouse while in Live Preview.

**Comment by lukemt** [comment]
Yeah, it will also be reordering checkboxes when you edit existing files. Like I said, as long as it is behind a setting, it should be fine.

I do have an alternative idea.

It is possible to track clicks with mouse events. You could attach an event listener to the document root and check whether the clicked upon element was a checkbox. I did something different but similar in the [opener plugin](https://github.com/aidan-gibson/obsidian-opener/), you could have a look there.

Then for the commands you could just add new commands that do both, toggle a checkbox and reorder afterwards. 

One could call the original commands like so:
`this.app.commands.executeCommandById("daily-notes:goto-prev")`.

If you enter app.commands into the dev console, you will find a list of available commands

**Comment by lukemt** [comment]
The relevant commands that i know of are toggle checkbox status and cycle bullet/checkbox

**Comment by lukemt** [comment]
Another idea: reorder only when strg-clicked or middle clicked but this is a little over the top too

**Comment by TakeADev** [comment]
> Yeah, it will also be reordering checkboxes when you edit existing files. Like I said, as long as it is behind a setting, it should be fine.
> 
> I do have an alternative idea.
> 
> It is possible to track clicks with mouse events. You could attach an event listener to the document root and check whether the clicked upon element was a checkbox. I did something different but similar in the [opener plugin](https://github.com/aidan-gibson/obsidian-opener/), you could have a look there.
> 
> Then for the commands you could just add new commands that do both, toggle a checkbox and reorder afterwards.
> 
> One could call the original commands like so: `this.app.commands.executeCommandById("daily-notes:goto-prev")`.
> 
> If you enter app.commands into the dev console, you will find a list of available commands

Ok I like this idea. Let me mess around with this as well as the other feedback you had on my PR and we'll get it going. 

Question: As this is really my first time attempting to contribute to open source, what's the best way to go about making the changes? Should I retract my PR and submit a new one?

**Comment by lukemt** [comment]
Ha, I apologize for jumping on it like that 😉!

If you have any questions, feel free to ask.

In this case I would probably start a new branch from main and close the PR and create a new one. 

But continuing here would probably work just as well. 

The main reason why I would start anew would be to keep the commit history clean, however, depending on the merge strategy this might not even apply.

And isn't a really serious project it's just a fun toy, so don't worry too much.

**Comment by TakeADev** [comment]
No worries at all! I appreciate the feedback. And thank you for the help!

Sounds good though I'll get working on it :D

**Comment by jenningsb2** [comment]
Hey! I'm just now coming across this plugin. I hope it's still being worked on! 

I just wanted to say that automatic updates would be a great addition.
